database/postgres/dal/refresh.py

```diff
@@ -0,0 +1,121 @@
+from sqlalchemy.orm import Session
+from sqlalchemy.exc import SQLAlchemyError
+from typing import Optional
+from datetime import datetime
+
+from database.models.refresh_token_tracking import RefreshTokenTracking
+from database.models.token_blocklist import TokenBlocklist
+from database.postgres.dto.refresh import refreshDTO
+
+class RefreshTokenDAL:
+    def __init__(self, db_session: Session):
+        self.db_session = db_session
+
+    def save_refresh_token(self, refresh_dto: refreshDTO, last_ip: str, last_device: str, nonce: str) -> Optional[int]:
```

Comment:
Why do we have a bunch of strings in method args if we could easily move them to a DTO?